shell/platform/embedder/tests/embedder_vk_unittests.cc

```diff
@@ -0,0 +1,123 @@
+
+namespace {
+
+struct VulkanProcInfo {
+  decltype(vkGetInstanceProcAddr)* get_instance_proc_addr = nullptr;
+  decltype(vkGetDeviceProcAddr)* get_device_proc_addr = nullptr;
+  decltype(vkQueueSubmit)* queue_submit_proc_addr = nullptr;
+  bool did_call_queue_submit = false;
+};
+
+static_assert(std::is_trivially_destructible_v<VulkanProcInfo>);
+
+VulkanProcInfo g_vulkan_proc_info;
+
+VkResult QueueSubmit(VkQueue queue,
+                     uint32_t submitCount,
+                     const VkSubmitInfo* pSubmits,
+                     VkFence fence) {
+  FML_DCHECK(g_vulkan_proc_info.queue_submit_proc_addr != nullptr);
+  g_vulkan_proc_info.did_call_queue_submit = true;
+  return g_vulkan_proc_info.queue_submit_proc_addr(queue, submitCount, pSubmits,
+                                                   fence);
+}
+
+PFN_vkVoidFunction GetDeviceProcAddr(VkDevice device, const char* pName) {
+  FML_DCHECK(g_vulkan_proc_info.get_device_proc_addr != nullptr);
+  if (strcmp(pName, "vkQueueSubmit") == 0) {
+    g_vulkan_proc_info.queue_submit_proc_addr =
+        reinterpret_cast<decltype(vkQueueSubmit)*>(
+            g_vulkan_proc_info.get_device_proc_addr(device, pName));
+    return reinterpret_cast<PFN_vkVoidFunction>(QueueSubmit);
+  }
+  return g_vulkan_proc_info.get_device_proc_addr(device, pName);
+}
+
+PFN_vkVoidFunction GetInstanceProcAddr(VkInstance instance, const char* pName) {
+  FML_DCHECK(g_vulkan_proc_info.get_instance_proc_addr != nullptr);
+  if (strcmp(pName, "vkGetDeviceProcAddr") == 0) {
+    g_vulkan_proc_info.get_device_proc_addr =
+        reinterpret_cast<decltype(vkGetDeviceProcAddr)*>(
+            g_vulkan_proc_info.get_instance_proc_addr(instance, pName));
+    return reinterpret_cast<PFN_vkVoidFunction>(GetDeviceProcAddr);
+  }
+  return g_vulkan_proc_info.get_instance_proc_addr(instance, pName);
+}
+
+template <typename T, typename U>
+struct CheckSameSignature : std::false_type {};
+
+template <typename Ret, typename... Args>
+struct CheckSameSignature<Ret(Args...), Ret(Args...)> : std::true_type {};
+
+static_assert(CheckSameSignature<decltype(GetInstanceProcAddr),
+                                 decltype(vkGetInstanceProcAddr)>::value);
+static_assert(CheckSameSignature<decltype(GetDeviceProcAddr),
+                                 decltype(vkGetDeviceProcAddr)>::value);
+static_assert(
+    CheckSameSignature<decltype(QueueSubmit), decltype(vkQueueSubmit)>::value);
+}  // namespace
+
+TEST_F(EmbedderTest, CanSwapOutVulkanCalls) {
+  auto& context = GetEmbedderContext(EmbedderTestContextType::kVulkanContext);
+  fml::AutoResetWaitableEvent latch;
+  context.AddIsolateCreateCallback([&latch]() { latch.Signal(); });
+  EmbedderConfigBuilder builder(context);
+  builder.SetVulkanRendererConfig(
+      SkISize::Make(1024, 1024),
+      [](void* user_data, FlutterVulkanInstanceHandle instance,
+         const char* name) -> void* {
+        if (strcmp(name, "vkGetInstanceProcAddr") == 0) {
```

Comment:
strncmp might be a minor improvement but std::string_view has an operator== :)

Reply:
done (switched to strncmp)